exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillStatementImpl.java

```diff
@@ -204,7 +276,7 @@ public boolean isClosed() {
 
   @Override
   public int getMaxFieldSize() throws SQLException {
-    throwIfClosed();
+    throwIfTimedOutOrClosed();
```

Comment:
The throwIfTimedOutOrClosed() call is basically a wrapper around a sequential check for, first the timed-out state, followed by the check for the closed state. A timed-out query (i.e. statement/resultset) is already in a closed state, but we need to throw the correct exception (in this case, the timeout), which is why it was done like that. 
My understanding was that any execute and data fetch operations can throw timeoutExceptions. Are you suggesting that for such '_getter_' methods, only an AlreadyClosed exception needs to be thrown and not time out?